sda-commons-server-kafka/src/test/java/org/sdase/commons/server/kafka/KafkaPrometheusMonitoringIT.java

```diff
@@ -140,4 +137,197 @@ void shouldWriteHelpAndTypeToMetrics() {
+
+    String topic2 = "testTopicKafkaMetrics_2";
+    KAFKA.getKafkaTestUtils().createTopic(topic2, 1, (short) 1);
+
+    AutocommitMLS<Long, Long> longLongAutocommitMLS =
+        new AutocommitMLS<>(
+            record -> resultsLong.add(record.value()), new IgnoreAndProceedErrorHandler<>());
+    createMessageListener(topic, CONSUMER_1, longLongAutocommitMLS);
+
+    AutocommitMLS<Long, Long> longLongAutocommitMLS2 =
+        new AutocommitMLS<>(
+            record -> resultsLong2.add(record.value()), new IgnoreAndProceedErrorHandler<>());
+    createMessageListener(topic2, "consumer3", longLongAutocommitMLS2);
+
+    MessageProducer<Long, Long> producer = registerProducer(topic, PRODUCER_1);
+    MessageProducer<Long, Long> producer2 = registerProducer(topic2, "producer3");
+
+    // pass in messages
+    producer.send(1L, 1L);
+    producer.send(2L, 2L);
+
+    producer2.send(1L, 1L);
+    producer2.send(2L, 2L);
+
+    attachPrometheusToMicrometer();
+
+    await()
+        .atMost(KafkaBundleConsts.N_MAX_WAIT_MS, MILLISECONDS)
+        .until(() -> resultsLong.size() == 2);
+
+    await()
+        .atMost(KafkaBundleConsts.N_MAX_WAIT_MS, MILLISECONDS)
+        .until(() -> resultsLong2.size() == 2);
+
+    ArrayList<Collector.MetricFamilySamples> collectorMetricSamplesErrorTotal =
+        Collections.list(
+            CollectorRegistry.defaultRegistry.filteredMetricFamilySamples(
+                s -> s.equals("kafka_producer_record_error_total")));
+
+    //    assert that metric exists
+    assertThat(collectorMetricSamplesErrorTotal).hasSize(1);
+
+    //    assert that samples for both producers are recorded
+    assertThat(collectorMetricSamplesErrorTotal.get(0).samples).hasSize(2);
+
+    List<Collector.MetricFamilySamples.Sample> samples =
+        collectorMetricSamplesErrorTotal.get(0).samples;
+
+    List<String> sampleLabelValueNames =
+        samples.stream().map(s -> s.labelValues.get(0)).collect(Collectors.toList());
+    assertThat(sampleLabelValueNames).contains("producer1", "producer3");
+
+    assertThatAllMicrometerMetricsExist();
+  }
+
+  private void assertThatAllMicrometerMetricsExist() {
+    List<Optional<String>> optionalMetricNamesMicrometer =
+        Collections.list(CollectorRegistry.defaultRegistry.metricFamilySamples()).stream()
+            .map(
+                metricFamilySamples ->
+                    metricFamilySamples.samples.stream().map(sample -> sample.name).findFirst())
+            .collect(Collectors.toList());
+
+    List<String> metricNamesMicrometer =
+        optionalMetricNamesMicrometer.stream()
+            .filter(o -> o.isPresent())
+            .map(name -> name.get())
+            .collect(Collectors.toList());
+
+    assertThat(metricNamesMicrometer).containsAll(getMetricsThatShouldExist());
+  }
+
+  private List<String> getMetricsThatShouldExist() {
+
+    ArrayList<String> result = new ArrayList<>();
+
+    result.add("kafka_producer_batch_split_rate");
```

Comment:
As a suggestion, you can use List.of("string1", "string2", ...)

Reply:
updated